Update README.md (#14)
apache/db-jdo-site: README.md

## Building the Site

The content and styling of the site is defined in the [AsciiDoc](https://asciidoc.org/) format. It is built using [Maven](https://maven.apache.org/). For details on publishing the site see section [Publishing the Site](#publishing-the-site).

The site can be built by calling `mvn clean compile`. This generates the HTML files in `target/site`.
Most of the site will work with the exception of the javadoc file downloads.
If needed, call `mvn package`. This copies the javadoc files to `target/site`.
The site can then be viewed by opening the local file `target/site/index.html` in a browser.

### Adding Javadoc

The site contains a packaged version of the JDO API javadoc. It can be updated as follows:

* Create the javadoc jar (e.g. jdo-api-3.2-javadoc.jar) in the db-jdo repository by calling `mvn clean install -Papache-release` in the api submodule.

> * Create a new folder under docs e.g. `src/main/resources/javadoc/api32`.

Reviewer: ```suggestion
* Create a new folder under in the javadoc resources directory, e.g. `src/main/resources/javadoc/api32`.
```

## Building the Site

> The content and styling of the site is defined in the [AsciiDoc](https://asciidoc.org/) format. It is built using [Maven](https://maven.apache.org/). For details on publishing the site see below.

Reviewer: Add a relative link to the referenced section?

```suggestion
The content and styling of the site is defined in the [AsciiDoc](https://asciidoc.org/) format. It is built using [Maven](https://maven.apache.org/). For details on publishing the site see section [Publishing the Site](#publishing-the-site).
```

 * The AsciiDoc sources can be found in `src/main/asciidoc`.
 * The website menu is defined in `src/main/template`.
 * The converter for migrating the old HTML files to AsciiDoc can be found in `src/main/java`.
 * Additional pre-compiled resources are located in `src/main/resources`.

Contributions to this repository follow the default [GitHub workflow](https://guides.github.com/introduction/flow/)
using [forks](https://guides.github.com/activities/forking/).

To contribute changes, you can follow these steps:

 * Adapt the AsciiDoc files in `src/main/asciidoc` or the website menu in  `src/main/template`.

>  * Build the site (see [above](#building-the-site) and verify the generated website by viewing `target/site/index.html` locally with a web browser.

Reviewer: ```suggestion
 * Build the site (see [above](#building-the-site)) and verify the generated website by viewing `target/site/index.html` locally with a web browser.
```

To contribute changes, you can follow these steps:

 * Adapt the AsciiDoc files in `src/main/asciidoc` or the website menu in  `src/main/template`.

>  * Call `mvn clean install` to build the site and verify the generated website by viewing `target/index.html` locally with a web browser.

Reviewer: With this new call, the build artifacts are automatically copied to the `docs` directory, which is not part of the `.gitignore`. As a result, such changes could (accidentally) be included in the commit if everything in the workspace is added to the staging area (i.e. by using `git add .`, `git add -u`, or ` git commit -a`).

For this reason, I would suggest still keeping `mvn clean compile` in this section or at least mentioning that the build artifacts are copied to `docs` and should be avoided when adding changes to the staging area.

 * Commit the source changes (not the build artifacts) and open a pull request.

### TODO
 * If you find any issues please provide a PR or [create a JIRA ticket](https://issues.apache.org/jira/projects/JDO/issues/?filter=allopenissues)
 

> ### PUBLISHING THE SITE

Reviewer: I would suggest using title case for the section headings.

```suggestion
### Publishing the Site
```

After changes have been made to the sources in the `db-jdo-site/src/main/asciidoc` or `db-jdo-site/src/main/template` directory, changes will be published automatically to the live web site by simply pushing changes to the master branch of the repository. The process is as follows:

1. Pushing changes to the master branch invokes the post-push script in `db-jdo-site/.github/workflows/deploy-site.yml` which builds the site in `db-jdo-site/target/` via `mvn clean install`.

1. If the build is successful, the files are copied from `db-jdo-site/target/` to `db-jdo-site/docs/`. The `db-jdo-site/docs` directory is checked to see if any changes have been made.

> 1. It is possible that the user made changes to `db-jdo-site/src/main/asciidoc/` and also compiled and pushed them to `db-jdo-site/docs/`). In this case, proceed to step 5.

Reviewer: Markdown also allows for more complex lists. You could add this step and the following steps as sub-entries in the list to make it easier to read. As an example:

1. Stuff
2. More stuff
a. option 1
b. option 2
3. Finally even more stuff
